pkg/api/validation/validation.go

```diff
@@ -234,6 +235,14 @@ func ValidatePositiveQuantity(value resource.Quantity, fieldName string) errs.Va
 	return allErrs
 }
 
+func ValidateImmutableField(old, new interface{}, fieldName string) errs.ValidationErrorList {
```

Comment:
@brendandburns 
From my perspective, returning `errs.ValidationErrorList` is a better choice, just as `ValidatePositiveField` and `ValidatePositiveQuantity` do. Those three validation functions can indeed only return a single error, but there are lots of validation functions such as `ValidateLabelName` will return a list of error, keep all validation functions consistently return `errs.ValidationErrorList｀could make all invocation statements consistent.